### ISSUE TITLE
solidity 0.8.0，truffle-flattener，packaging sol file problem

### ISSUE THREAD

**Issue by dengdeng95** [nomiclabs/truffle-flattener#83]
Error: Could not parse G:\project\node_modules\@openzeppelin\contracts\token\ERC1155\ERC1155.sol for extracting its imports: ParserError: missing ';' at '{' (177:18)

**Comment by lancekrogers** [comment]
I'm running into the same problem

**Comment by dengdeng95** [comment]
> I'm running into the same problem
npx github:fedeb95/truffle-flattener
use this to solve the problem.

**Comment by naftalimurgor** [comment]
Hello @dengdeng95 I'm experiencing a similar issue with soldity@0.8.0 Hardhat has a flattened too. I'll try using the hardhat at the moment.

**Comment by naftalimurgor** [comment]
@dengdeng95 @lancekrogers  I managed to resolve this issue like this:
1. `npx hardhat` to create a new hardhat project, gives a menu to create a project from the command line, very much similar to a truffle project
2. add @openzeppelin contracts using npm like this:
```bash
npm install @openzeppelin/contracts@4.3.0
```
3. copy your old contracts to `contracts/` folder
4. run `npx hardhat flatten contracts/NRT.sol > flattened.sol`
The flattened contract shall be output file named `flattened.sol` in the project root
The contract names I picked are just arbitrary names I picked, you can use your filenames for the contracts and the output file after the `>` symbol.
That's how I resolved  this issue

**Comment by dengdeng95** [comment]
> npx github:fedeb95/truffle-flattener

**Comment by dengdeng95** [comment]
> @dengdeng95 @lancekrogers I managed to resolve this issue like this:
> 
> 1. `npx hardhat` to create a new hardhat project, gives a menu to create a project from the command line, very much similar to a truffle project
> 2. add @OpenZeppelin contracts using npm like this:
> 
> ```shell
> npm install @openzeppelin/contracts@4.3.0
> ```
> 
> 3. copy your old contracts to `contracts/` folder
> 4. run `npx hardhat flatten contracts/NRT.sol > flattened.sol`
>    The flattened contract shall be output file named `flattened.sol` in the project root
>    The contract names I picked are just arbitrary names I picked, you can use your filenames for the contracts and the output file after the `>` symbol.
>    That's how I resolved  this issue

npx github:fedeb95/truffle-flattener
This command is ok!

**Comment by caffeinum** [comment]
This is fixed in v1.6.0 release:

<img width="286" alt="image" src="https://user-images.githubusercontent.com/1909384/160137191-9ec4d264-540b-4f26-b84b-7c2aa4f0af7b.png">
```bash
└─┬ truffle-flattener@1.6.0
  └── @solidity-parser/parser@0.14.1
```
